core/src/lib.rs

```diff
@@ -44,19 +45,33 @@ pub enum ValueType {
 }
 
 impl ValueType {
-    pub fn to_edn_value(&self) -> edn::Value {
+    pub fn to_edn_value(self) -> edn::Value {
         match self {
-            &ValueType::Ref => values::DB_TYPE_REF.clone(),
-            &ValueType::Boolean => values::DB_TYPE_BOOLEAN.clone(),
-            &ValueType::Instant => values::DB_TYPE_INSTANT.clone(),
-            &ValueType::Long => values::DB_TYPE_LONG.clone(),
-            &ValueType::Double => values::DB_TYPE_DOUBLE.clone(),
-            &ValueType::String => values::DB_TYPE_STRING.clone(),
-            &ValueType::Keyword => values::DB_TYPE_KEYWORD.clone(),
+            ValueType::Ref => values::DB_TYPE_REF.clone(),
+            ValueType::Boolean => values::DB_TYPE_BOOLEAN.clone(),
+            ValueType::Instant => values::DB_TYPE_INSTANT.clone(),
+            ValueType::Long => values::DB_TYPE_LONG.clone(),
+            ValueType::Double => values::DB_TYPE_DOUBLE.clone(),
+            ValueType::String => values::DB_TYPE_STRING.clone(),
+            ValueType::Keyword => values::DB_TYPE_KEYWORD.clone(),
         }
     }
 }
 
+impl fmt::Display for ValueType {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", match *self {
+            ValueType::Ref =>     "db.type/ref",
```

Comment:
Is there a reason you're not including the leading colon?  I don't feel strongly but it might be nice if `x.to_string() == x.to_edn_value().to_string()`.